bucket_map/src/bucket_storage.rs

```diff
@@ -335,11 +338,12 @@ impl<O: BucketOccupied> BucketStorage<O> {
     }
 
     /// allocate a new memory mapped file of size `bytes` on one of `drives`
-    fn new_map(drives: &[PathBuf], bytes: u64, stats: &BucketStats) -> (MmapMut, PathBuf) {
+    fn new_map(drives: &[PathBuf], bytes: u64, stats: &BucketStats) -> (MmapMut, PathBuf, u128) {
         let mut measure_new_file = Measure::start("measure_new_file");
         let r = thread_rng().gen_range(0..drives.len());
         let drive = &drives[r];
-        let pos = format!("{}", thread_rng().gen_range(0..u128::MAX),);
+        let file_random = thread_rng().gen_range(0..u128::MAX);
```

Comment:
One question, for my own understanding, what if `file_random` has already been used? Will this overwrite the existing file?

Reply:
Presumably the creation of the file would fail since it would be trying to be opened by 2 file open calls concurrently. The chances of us getting the same file twice are not very high.